hipchat/oauth.go

```diff
@@ -0,0 +1,108 @@
+package hipchat
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// ClientCredentials represents the OAuth2 client ID and secret for an integration
+type ClientCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+// OAuthAccessToken represents a newly created Hipchat OAuth access token
+type OAuthAccessToken struct {
+	AccessToken string `json:"access_token"`
+	ExpiresIn   uint32 `json:"expires_in"`
+	GroupID     uint32 `json:"group_id"`
+	GroupName   string `json:"group_name"`
+	Scope       string `json:"scope"`
+	TokenType   string `json:"token_type"`
+}
+
+// CreateClient creates a new client from this OAuth token
+func (t *OAuthAccessToken) CreateClient() *Client {
+	return NewClient(t.AccessToken)
+}
+
+// GetAccessToken returns back an access token for a given integration's client ID and client secret
```

Comment:
Please add the link to the api documentation in the comment, like other methods of the library.

```go
// GetAccessToken returns back an access token for a given integration's client ID and client secret
//
//  HipChat api docs : https://www.hipchat.com/docs/apiv2/method/generate_token
```